azkaban-web-server/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.commons.dbutils.ResultSetHandler;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class ImageVersionDaoImpl implements ImageVersionDao {
+
+  private static final Logger log = LoggerFactory.getLogger(ImageVersionDaoImpl.class);
+
+  private final DatabaseOperator databaseOperator;
+
+  private final ImageTypeDao imageTypeDao;
+
+  private static String INSERT_IMAGE_VERSION_QUERY =
+      "insert into image_versions ( path, description, version, type_id, state, "
+          + "release_tag, created_on, created_by) "
+          + "values (?, ?, ?, ?, ?, ?, ?, ?)";
+  private static String SELECT_IMAGE_VERSION_BASE_QUERY = "select iv.id, iv.path, iv.description, "
+      + "iv.version, it.type, iv.state, iv.release_tag, iv.created_on, iv.created_by, iv.modified_on, "
+      + "iv.modified_by from image_versions iv, image_types it where it.id = iv.type_id";
+
+  @Inject
+  public ImageVersionDaoImpl(DatabaseOperator databaseOperator, ImageTypeDao imageTypeDao) {
+    this.databaseOperator = databaseOperator;
+    this.imageTypeDao = imageTypeDao;
+  }
+
+  @Override
+  public int createImageVersion(ImageVersion imageVersion) {
+    ImageType imageType = imageTypeDao.getImageTypeByType(imageVersion.getType())
+        .orElseThrow(() -> new RuntimeException("Unable to fetch image type metadata. Invalid "
+            + "image type : "+imageVersion.getType()));
+    final SQLTransaction<Long> insertAndGetSpaceId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_IMAGE_VERSION_QUERY, imageVersion.getPath(), imageVersion.getDescription(),
+          imageVersion.getVersion(), imageType.getId(), imageVersion.getState().getStateValue(),
+          imageVersion.getReleaseTag(), currentTime, imageVersion.getCreatedBy());
+      transOperator.getConnection().commit();
+      return transOperator.getLastInsertId();
+    };
+
+    int imageVersionId = 0;
+    try {
+      /* what will happen if there is a partial failure in
+         any of the below statements?
+         Ideally all should happen in a transaction */
+      imageVersionId = databaseOperator.transaction(insertAndGetSpaceId).intValue();
+    } catch (SQLException e) {
+      log.error("Unable to create the image version metadata", e);
+    }
+    return imageVersionId;
+  }
+
+  @Override
+  public List<ImageVersion> getImageVersion(RequestContext requestContext) throws ImageMgmtException {
+    List<ImageVersion> imageVersions = new ArrayList<>();
+    try {
+      StringBuilder queryBuilder = new StringBuilder(SELECT_IMAGE_VERSION_BASE_QUERY);
+      if(requestContext.getParams().containsKey(ImageMgmtConstants.IMAGE_TYPE)) {
+        queryBuilder.append(" AND ");
+        queryBuilder.append(" it.type = '"+requestContext.getParams().get(ImageMgmtConstants.IMAGE_TYPE)+"'");
```

Comment:
Can you please use ? as a place holder in query and pass parameter in query method like https://github.com/azkaban/azkaban/blob/497d4cdb9520781531e2881cc6771d6bcf45be49/azkaban-common/src/main/java/azkaban/executor/FetchActiveFlowDao.java#L165?

Reply:
Done.